src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3795,13 +3863,120 @@ export class StudyViewPageStore
         );
     }
 
+    @observable private sampleIdToGenePanelId: {
+        [sampleId: string]: string;
+    } = {};
+
+    readonly genePanelData = remoteData<{
+        [uniqueKey: string]: MobxPromise<ClinicalDataCountSummary[]>;
+    }>({
+        await: () => [
+            this.selectedSamples,
+            this.molecularProfiles,
+            this.molecularProfileNameSet,
+        ],
+        invoke: async () => {
+            let sampleIds = this.selectedSamples.result.map(s => s.sampleId);
+            const molecularProfileIds = this.molecularProfiles.result.map(
+                mp => mp.molecularProfileId
+            );
+            const molecularProfileNameSet = this.molecularProfileNameSet
+                .result!;
+            const genePanelData: {
+                [uniqueKey: string]: MobxPromise<ClinicalDataCountSummary[]>;
+            } = {};
+
+            for (const profileId of Object.keys(molecularProfileNameSet)) {
+                const uniqueKey = getGenePanelChartUniqueKey(profileId);
+                genePanelData[uniqueKey] = remoteData<
```

Comment:
probably not a good idea to return remoteData inside a remoteData

Reply:
i guess we're doing this down below.  hmmm

Reply:
Technically `genePanelData` (the outer remoteData) isn't really remote data. It just needs to be a promise since I'm waiting for some other promises to complete

Reply:
i guess it allows us to track the individual loading state of the nested promises.  do we actually use that ability?

Reply:
If you're talking about the nested remote data, then I need each return value to be its own `MobxPromise` which is why I use remote data. See line 3977 in `StudyViewPageStore`: `return this.genePanelData.result[uniqueKey]`. So I access and return each promise individually